pkg/policy/policies.go

```diff
@@ -213,6 +268,489 @@ func (ps *Policies) FilterableInUserlandMap() map[*Policy]int {
+	// 2. comm_filter    string_filter_t, eq_t
+
+	for k, v := range strEqualities {
+		byteStr := make([]byte, maxBpfStrFilterSize)
+		copy(byteStr, k)
+		keyPointer := unsafe.Pointer(&byteStr[0])
+
+		eqVal := make([]byte, equalityValueSize)
+		valuePointer := unsafe.Pointer(&eqVal[0])
+
+		binary.LittleEndian.PutUint64(eqVal[0:8], v.equalInPolicies)
+		binary.LittleEndian.PutUint64(eqVal[8:16], v.equalitySetInPolicies)
+
+		bpfMap, ok := ps.bpfFilterMaps[innerMapName]
+		if !ok {
+			return errfmt.Errorf("bpf map not found: %s", innerMapName)
+		}
+		if err := bpfMap.Update(keyPointer, valuePointer); err != nil {
+			return errfmt.WrapError(err)
+		}
+	}
+
+	return nil
+}
+
+const (
+	maxBpfBinPathSize = 256 // maximum binary path size supported by BPF (MAX_BIN_PATH_SIZE)
+	bpfBinFilterSize  = 264 // the key size of the BPF binary filter map entry
+)
+
+func (ps *Policies) updateBinaryFilterBPF(binEqualities map[filters.NSBinary]equality, innerMapName string) error {
+	// BinaryNS equality
+	// 1. binary_filter  binary_t, eq_t
+
+	for k, v := range binEqualities {
+		if len(k.Path) > maxBpfBinPathSize {
+			return filters.InvalidValue(k.Path)
+		}
+		binBytes := make([]byte, bpfBinFilterSize)
+		if k.MntNS == 0 {
+			// if no mount namespace given, bpf map key is only the path
+			copy(binBytes, k.Path)
+		} else {
+			// otherwise, key is composed of the mount namespace and the path
+			binary.LittleEndian.PutUint32(binBytes, k.MntNS)
+			copy(binBytes[4:], k.Path)
+		}
+		keyPointer := unsafe.Pointer(&binBytes[0])
+
+		eqVal := make([]byte, equalityValueSize)
+		valuePointer := unsafe.Pointer(&eqVal[0])
+
+		binary.LittleEndian.PutUint64(eqVal[0:8], v.equalInPolicies)
+		binary.LittleEndian.PutUint64(eqVal[8:16], v.equalitySetInPolicies)
+
+		bpfMap, ok := ps.bpfFilterMaps[innerMapName]
+		if !ok {
+			return errfmt.Errorf("bpf map not found: %s", innerMapName)
+		}
+		if err := bpfMap.Update(keyPointer, valuePointer); err != nil {
+			return errfmt.WrapError(err)
+		}
+	}
+
+	return nil
+}
+
+type procInfo struct {
+	newProc        bool
+	followPolicies uint64
+	mntNS          uint32
+	binaryBytes    [maxBpfBinPathSize]byte
+	binNoMnt       uint32
+}
+
+// TODO: Should ProcInfoMap be cleared when a new version is created?
+// Or should it be versioned too?
+func populateProcInfoMap(bpfModule *bpf.Module, binEqualities map[filters.NSBinary]equality) error {
+	procInfoMap, err := bpfModule.GetMap(ProcInfoMap)
+	if err != nil {
+		return errfmt.WrapError(err)
+	}
+
+	binsProcs, err := proc.GetAllBinaryProcs()
+	if err != nil {
+		return errfmt.WrapError(err)
+	}
+
+	// populate proc info map with binaries to track
+	for bin := range binEqualities {
+		procs := binsProcs[bin.Path]
+		for _, p := range procs {
+			binBytes := make([]byte, maxBpfBinPathSize)
+			copy(binBytes, bin.Path)
+			binBytesCopy := (*[maxBpfBinPathSize]byte)(binBytes)
+			procInfo := procInfo{
+				newProc:        false,
+				followPolicies: 0,
```

Comment:
These assumption of default values are no longer correct if we update the policies dynamically.
Since it was assumed we call this function on init, it was safe to say that this is not a newProc (started after tracee) and that not match occured (thus follow can be set to false for this process).

We can comment it here as a todo and handle later